### ISSUE TITLE
Datagrid Single Row Action no longer works when ResizeSensor is used by other elements

### ISSUE THREAD

**Issue by whizkidwwe1217** [vmware/clarity#3799]
## Describe the bug

I've been using the `ResizeSensor` which is a part of [css-element-queries](https://github.com/marcj/css-element-queries) library to detect the changes in dimension of any HTML element. I don't have any problems with it prior to Clarity version 2.2.0 but when I updated my dependencies to the latest Clarity version, the data grid component's single row action, which I think is now using the new smart popover, it's not popping up.

Resize sensor is disabled:
![image](https://user-images.githubusercontent.com/1977391/64543254-79c3c780-d357-11e9-96f8-4a50611732ab.png)

Resize sensor is enabled and applied to the _toolbar_ element:
![image](https://user-images.githubusercontent.com/1977391/64543879-94e30700-d358-11e9-93f0-055309837374.png)

## How to reproduce
Sample code that uses the ResizeSensor (I created a directive for this):
```html
<div (elementResized)="onResize($event)">
    <button class="btn">Button 1</button>
    <button class="btn">Button 1</button>
</div>

<clr-datagrid>
    ...datagrid definition here
</clr-datagrid>
```

Use this StackBlitz link to see the issue:  https://stackblitz.com/edit/clarity-light-theme-v2-yrfkcg
1. Click on the `Toggle Resize Sensor` button to enable/disable the resize sensor.
2. Click on the single action row button.
![V2Kq5UGDfm](https://user-images.githubusercontent.com/1977391/64544190-25b9e280-d359-11e9-8a30-2c601205de99.gif)

## Expected behavior

The datagrid single row action should show up when clicked. If you have an alternative that I can use for detecting the size changes in an element, then I think I get rid of the `ResizeSensor` library.

## Versions

**App**

* Angular: [e.g. 6] 8
* Node: [e.g. 8.10.0] 10 or 11
* Clarity: [e.g. 0.12.5] 2.2.0

**Device:**

* Type: [e.g. MacBook] Dell XPS 13
* OS: [e.g. iOS] Windows 10
* Browser [e.g. chrome, safari] Chrome, Edge Chromium, Edge, Firefox, IE11
* Version [e.g. 22] Latest

## Additional notes

Column filters and column hide/show toggle seem fine. I think only the single action row button is not working.

**Comment by hippee-lee** [comment]
Hi There, 

I did a bit of investigation on this. One thing I noticed is that I can only replicate the issue as described wen using the stackblitz code editor and preview pane in the same tab. The popover works as expected when `Toggle Resize Sensor` is on in the full app view: https://clarity-light-theme-v2-yrfkcg.stackblitz.io/

Is this the case for you as well?

**Comment by gnomeontherun** [comment]
First, what are you using this ResizeSensor for exactly? I'm not sure why you are using it but since this is a third party service it is unclear to me if this is an issue on our end or theirs.

**Comment by whizkidwwe1217** [comment]
@gnomeontherun It doesn't work in the full app view for me either. When the button is green, that means ResizeSensor is being used. I made a toggle button so I can see the difference when using and not using the ResizeSensor.
I built a custom toolbar component similar to Microsoft's Office UI Fabric [CommandBar component](https://developer.microsoft.com/en-us/fabric/#/controls/web/commandbar) and Blueprint's [Overflow list](https://blueprintjs.com/docs/#core/components/overflow-list) component. The component can contain several buttons/toolbar items. If the width of the toolbar is enough to display all the buttons then it'll display all of them. But, if the width is smaller than the total width of all the buttons, those buttons that aren't visible are added to the overflow button.

The reason why I used this library is because I used it to listen to the changes in the dimension of an element. With that, I can recalculate the width of the toolbar and the total width of all its buttons. Then when I got the right calculated widths, I can now decide whether I should make the buttons visible in the toolbar or hide it in the overflow button.

To summarize, my goal is to listen to the changes in the HTML element's dimensions. I can't use the html resize event since it fires only when resizing the browser window.

**Comment by whizkidwwe1217** [comment]
Here's what it looks like in real-world app:

When the width of the page is wider:
![image](https://user-images.githubusercontent.com/1977391/64577852-39446800-d3b0-11e9-976a-0a3eedaefabe.png)

When the width is tight:
![image](https://user-images.githubusercontent.com/1977391/64577932-76a8f580-d3b0-11e9-8bef-3744b0f654d1.png)

**Comment by whizkidwwe1217** [comment]
@gnomeontherun I've done some research and came up with an alternative solution. I totally got rid of the ResizeSensor library and used Google Dev's [ResizeObserver](https://developers.google.com/web/updates/2016/10/resizeobserver) instead. Tested it, used some polyfills for other browsers, and now I've got the single action row working again! Anyway, thanks for looking into this issue and I can now close this.

**Comment by hippee-lee** [comment]
Awesome. fwiw - I think `ResizeObserver` is part of the WebAPI and not exclusive to Chrome/google https://drafts.csswg.org/resize-observer-1/ and https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver.
